Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/AuditSearchBean.java

```diff
@@ -0,0 +1,269 @@
+ *       and/or other materials provided with the distribution.
+ *     * Neither the name of the United States Government nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.admingui.managed;
+
+import gov.hhs.fha.nhinc.admingui.constant.NavigationConstant;
+import gov.hhs.fha.nhinc.admingui.event.model.Audit;
+import gov.hhs.fha.nhinc.admingui.services.AuditService;
+import gov.hhs.fha.nhinc.admingui.services.impl.AuditServiceImpl;
+import gov.hhs.fha.nhinc.admingui.util.RemoteOrganizationIdentifier;
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.faces.bean.ManagedBean;
+import javax.faces.bean.SessionScoped;
+
+/**
+ *
+ * @author achidamb
+ */
+@ManagedBean(name = "auditSearchBean")
+@SessionScoped
+public class AuditSearchBean {
+
+    private String userId;
+    private Map<String, String> remoteHcidList;
+    private Date eventStartDate;
+    private Date eventEndDate;
+    private ArrayList<String> eventTypeList;
+    private int activeIndex = 0;
+    private String outcomeIndicator;
+    private ArrayList<String> selectedRemoteHcidList;
+    private ArrayList<String> selectedEventTypeList;
+    private Map<String, String> eventOutcomeIndicatorList;
+    private ArrayList<Audit> auditRecordList;
+    private String auditMessage;
+    private boolean auditFound;
+    private String messageId;
+    private String RelatesTo;
+
+    public AuditSearchBean() {
+        setEventTypeList(populateEventTypeList());
+        setRemoteHcidList(populateRemoteOrgHcid());
+        setEventOutcomeIndicatorList(populateEventOutcomeIndicatorValues());
+    }
+
+    public void searchAudit() {
+        AuditService impl = new AuditServiceImpl();
+        this.auditRecordList = impl.createMockAuditRecord();
+        if (this.auditRecordList.isEmpty() && this.auditRecordList == null) {
+            this.auditMessage = "Audit Records not found";
```

Comment:
We should always do a null check before doing an empty check. Better yet you can use NullChecker.isNotNullish here.